app/services/resource_recommendation.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+class ResourceRecommendation
+  def initialize(moment)
+    @moment = moment
+    @moment_keywords = []
+    @matched_resources = []
+  end
+
+  def get_keywords(array)
+    array.each do |item|
+      @moment_keywords.push(item['name'].split,
+                            html_clean(item['description']).split)
+    end
+  end
+
+  def modify_keywords
+    @moment_keywords = @moment_keywords.flatten
+    @moment_keywords.each do |keyword|
+      keyword.gsub!(%r{([_@#!%()\-=;><,{}\~\[\]\./\?\"\*\^\$\+\-]+)}, '')
+    end
+    @moment_keywords = @moment_keywords.map(&:downcase)
+  end
+
+  def match_keywords
+    all_resources.each do |resource|
+      resource_tags = resource['tags'].map do |tag|
+        tag.split('_')
+      end
+      unless (resource_tags.flatten & @moment_keywords).empty?
+        @matched_resources.push(resource)
+      end
```

Comment:
```suggestion
    all_resources.select do |resource|
      resource_tags = resource['tags'].map { |tag| tag.split('_') }
      
      (resource_tags.flatten & @moment_keywords).any?
```

There is an Array method called `select` that keeps items in an array only if the block returns `true`.

This eliminates the need for a conditional and to push to an external array. the returning value will be you matched resources.

https://ruby-doc.org/core-2.7.0/Array.html#method-i-select